projects/dendrites/permutedMNIST/hyperparameter_search/hyperparameters_dendrites_with_si_figures.py

```diff
@@ -0,0 +1,104 @@
+#
+#  This program is free software you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see htt"://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+import os
+
+import matplotlib.gridspec as gridspec
+import matplotlib.pyplot as plt
+import pandas as pd
+import ptitprince as pt
+import seaborn as sns
+
+sns.set(style="ticks", font_scale=1.3)
+
+
+def hyperparameter_dendrites_wit_si_search_panel():
+    """
+    Plots a 3 panels figure on 1 rows x 3 columns
+    Rows contains figures representing hyperparameters search for 10
+    permutedMNIST tasks resulting from hyperparameter_search.py config file.
+    Columns 1 is the number of dendritic segments, columns 2 the activation
+    sparsity and column 3 the weight sparsity.
+    """
+
+    df_path1 = f"{experiment_folder}si_centroid_hp_10_lasttask.csv"
+    df1 = pd.read_csv(df_path1)
+
+    df_path2 = f"{experiment_folder}si_centroid_hp_10_control_lasttask.csv"
+    df2 = pd.read_csv(df_path2)
+
+    df1 = df1[["Activation sparsity", "FF weight sparsity", "Num segments",
+               "Accuracy"]]
+    df2 = df2[["Activation sparsity", "FF weight sparsity", "Num segments",
+               "Accuracy"]]
+    df1["condition"] = "dendrite_and_ff"
+    df2["condition"] = "ff_only"
+    df = pd.concat([df1, df2])
+
+    # Figure 1 'Impact of the different hyperparameters on performance
+    # full cross product of hyperparameters
+    gs = gridspec.GridSpec(1, 1)
+    fig = plt.figure(figsize=(5, 5))
+
+    ax1 = fig.add_subplot(gs[0, 0])
+
+    x1 = "Num segments"
+    dhue = "condition"
+    y = "Accuracy"
+    ort = "v"
+    pal = sns.color_palette(n_colors=9)
+    sigma = 0.2
+    fig.suptitle(
+        "Impact of the number of segments with SI on performance", fontsize=12
+    )
+
+    pt.RainCloud(x=x1, y=y, hue=dhue, data=df, palette=pal, bw=sigma, width_viol=0.6,
+                 ax=ax1, orient=ort, move=0.2, pointplot=True, alpha=0.65)
+    # pt.RainCloud(x=x1, y=y, data=df, palette=pal, bw=sigma, width_viol=0.6,
+    #              ax=ax1, orient=ort, move=0.2, pointplot=True, alpha=0.65)
+
+    ax1.set_ylabel("Mean accuracy", fontsize=16)
+    ax1.set_xlabel("Number of dendritic segments", fontsize=16)
+
+    # Add 10 tasks label
+    # plt.figtext(-0.02, 0.7, "10 TASKS", fontsize=16)
+
+    # fig.suptitle(
+    #     """Impact of the number of segments on \n 10-tasks
+    #     permuted MNIST performance""",
+    #     fontsize=16,
```

Comment:
We shouldn't have code that is commented out like this. If you don't need it, can you remove it?

(Also, note that I pushed to your branch, so you will need to pull first before you make changes)